django/contrib/auth/hashers.py

```diff
@@ -247,31 +262,43 @@ def encode(self, password, salt, iterations=None):
         hash = base64.b64encode(hash).decode('ascii').strip()
         return "%s$%d$%s$%s" % (self.algorithm, iterations, salt, hash)
 
-    def verify(self, password, encoded):
+    def decode(self, encoded):
         algorithm, iterations, salt, hash = encoded.split('$', 3)
         assert algorithm == self.algorithm
-        encoded_2 = self.encode(password, salt, int(iterations))
+        return {
+            'algorithm': algorithm,
+            'iterations': iterations,
+            'salt': salt,
+            'hash': hash,
+        }
+
+    def verify(self, password, encoded):
+        decoded_entry = self.decode(encoded)
+        assert decoded_entry['algorithm'] == self.algorithm
+        encoded_2 = self.encode(password,
+                                decoded_entry['salt'],
+                                int(decoded_entry['iterations']))
```

Comment:
Wouldn't it make sense to call `int` on `interations` alread in `decode`? From the naming alone I think it would make sense to move it into `decode`.

Reply:
I think it makes more sense to keep the dict data type consistent personally, but I can see the desire to have each element be its own native type. My concern is that down the line someone will access the element and not be able to tell if its a string/int/etc...

Reply:
That's fair. I think I was trying to keep the dict with a consistent datatype. I've added a commit to make the element an int.